### DISCUSSION TITLE
How do we select dropdown option using data-test-id in cypress

### DISCUSSION THREAD

**Discussion by ShehFahad** [cypress-io/cypress#15710] [Plugins] [answered]
I need to select option of `select` in cypress but using data-test-id, as option value and text is change in every environment. Only id is same for Test/Prod/QA.

We have option to pick using text by:
cy.get('select').select('Request builder modifier') 
but I need to pick using data-test-id instead.

![image](https://user-images.githubusercontent.com/54523045/113022044-a3447e00-919d-11eb-8cf1-7b0038a34545.png)

**Comment by bahmutov** [comment, marked as answer]
See how to do this in https://glebbahmutov.com/cypress-examples/6.8.0/recipes/select-value-by-id.html

```js
// let's find the option with the given test-id
const optionid = 'pick-oranges'
cy.get('[data-testid=select-example]')
  .find(`option[data-testid=${optionid}]`)
  .invoke('attr', 'value')
  .then((value) => {
    // now that we know the value of the <option>
    cy.get('[data-testid=select-example]').select(value)
  })
// confirm the selected option
cy.get('[data-testid=select-example] option:selected').should(
  'have.text',
  'Oranges',
)
```

**Reply by ShehFahad** [reply]
Thank you @bahmutov for you solution. It worked for me
